File: android/src/main/kotlin/com/exponea/style/TextViewStyle.kt
package com.exponea.style

import android.view.View
import android.widget.TextView
import com.exponea.PlatformSize
import com.exponea.asColor
import com.exponea.toTypeface

data class TextViewStyle(
    var visible: Boolean? = null,
    var textColor: String? = null,
    var textSize: String? = null,
    var textWeight: String? = null,
    var textOverride: String? = null
) {
    fun applyTo(target: TextView) {
        visible?.let {
            target.visibility = if (it) View.VISIBLE else View.GONE
        }
        textColor?.asColor()?.let {
            target.setTextColor(it)
        }
        PlatformSize.parse(textSize)?.let {
            target.setTextSize(it.unit, it.size)
        }
        textWeight?.let {
            target.setTypeface(target.typeface, toTypeface(it))
        }
        textOverride?.let { target.text = it }
    }
}

File: lib/src/data/model/text_view_style.dart
import 'package:exponea/exponea.dart';
import 'package:flutter/material.dart';

import '../util/Codable.dart';

class TextViewStyle extends Encodable {
  final bool? visible;
  final String? textColor;
  final String? textSize;
  final String? textWeight;
  final String? textOverride;

  TextViewStyle({
      this.visible,
      this.textColor,
      this.textSize,
      this.textWeight,
      this.textOverride});

  @override
  String toString() {
    return 'TextViewStyle{visible: $visible, textColor: $textColor,'
        'textSize: $textSize, textWeight: $textWeight,'
        'textOverride: $textOverride}';
  }

  @override
  Map<String, dynamic> encode() {
    return {
      'visible': visible,
      'textColor': textColor,
      'textSize': textSize,
      'textWeight': textWeight,
      'textOverride': textOverride
    };
  }

}
